### PR TITLE
Notebooks for ch02s02: Training Models PTL-12380 

### PR DESCRIPTION
**Pull request by jramcast**

### CODE REVIEW

File: models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    "\n",
+    "import pandas as pd\n",
+    "\n",
+    "import tensorflow as tf\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.metrics import classification_report"
+   ]
```
Comment (alxlenc) [inline models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb:32]: We should warn the student about the resulting warning.

File: models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+    "    \"BloodPressure\": 60.0,\n",
+    "    \"SkinThickness\": 35.0,\n",
+    "    \"Insulin\": 1.0,\n",
+    "    \"BMI\": 45.7,\n",
+    "    \"DiabetesPedigreeFunction\": 0.27,\n",
+    "    \"Age\": 20\n",
+    "}\n",
+    "\n",
+    "predictions = predict([diabetes_patient, no_diabetes_patient])\n",
+    "print(predictions)"
```
Comment (alxlenc) [inline models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb:296]: With a 79% model accuracy I always get `Diabetes, Diabetes` as the prediction result.

**Review by alxlenc — COMMENTED** [review]
On each of the notebooks, I would add a return to the course to continue with the exercise final instruction.

I only had issues with the TF predicctions, everything else worked.

File: models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb (hunk 1 above)
Comment (jramcast) [reply models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb:35]: Added a brief explanation about these warnings. Students can ignore them.

File: models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb (hunk 2 above)
Comment (jramcast) [reply models/training-models/diabetes-prediction/02-model-training-tensorflow.ipynb:298]: Fixed, now the prediction should be correct and reproducible.

**Review by alxlenc — APPROVED** [review]
LGTM

File: models/training-models/diabetes-prediction/01-data-exploration.ipynb [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    "data.hist(bins=20, figsize=(10, 8))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataset is evenly distributed except for some outliers in `BMI`, `Blood Pressure`, and `DiabetesPedigreeFunction`.\n",
+    "\n",
+    "Additionally, note the high number of `0` values in the `SKinThickness` and `Insulin` features.\n",
```
Comment (sffrench) [inline models/training-models/diabetes-prediction/01-data-exploration.ipynb:190]:
```suggestion
    "Additionally, note the high number of `0` values in the `SkinThickness` and `Insulin` features.\n",
```

File: models/training-models/diabetes-prediction/01-data-exploration.ipynb [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    "print(\"\\n == Percentage of 0's:\")\n",
+    "print(perZero)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The data set contains 227 zero values for `SkinThickness` and 374 zero values for `Insulin`.\n",
+    "Aproximately half of the cases have missing insuling values.\n",
```
Comment (sffrench) [inline models/training-models/diabetes-prediction/01-data-exploration.ipynb:267]:
```suggestion
    "Aproximately half of the cases have missing insulin values.\n",
```

File: models/training-models/diabetes-prediction/02-model-training-pytorch.ipynb [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+    "print(classification_report(y_test, y_predicted))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The trained model has an accuracy value of 79%.\n",
+    "\n",
+    "You can improve the score by retraining the model after more sophisticated data engineering or by tweaking the model hyper parameters."
```
Comment (sffrench) [inline models/training-models/diabetes-prediction/02-model-training-pytorch.ipynb:345]:
```suggestion
    "You can improve the score by retraining the model after more sophisticated data engineering or by tweaking the model's hyper parameters."
```

**Review by sffrench — COMMENTED** [review]
Quick review complete.
